File: library/src/main/java/com/jinkyumpark/library/region/region/Region.java
package com.jinkyumpark.library.region.region;

import com.jinkyumpark.library.region.regionDetail.RegionDetail;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Getter
@NoArgsConstructor @AllArgsConstructor @Builder

@Entity @Table
public class Region {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long regionId;

    @Column(nullable = false, length = 20) private String koreanName;
    @Column(length = 20) private String englishName;

    private String logo;
    private Integer data4LibCode;

    @Enumerated(value = EnumType.STRING)
    @Column(length = 10, nullable = false)
    private Country country;

    @OneToMany(mappedBy = "region", orphanRemoval = true)
    private List<RegionDetail> regionDetailList;

}

File: search/src/main/kotlin/com/jinkyumpark/search/offlineLibrary/provider/Region.kt
package com.jinkyumpark.search.offlineLibrary.provider

import javax.persistence.*

@Entity
@Table(
    name = "REGION",
    uniqueConstraints = [
        UniqueConstraint(name = "name_unique", columnNames = ["koreanName", "englishName"])
    ]
)
class Region {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Enumerated(EnumType.STRING) var country: Country? = null
    var koreanName: String? = null
    var englishName: String? = null

    @Column(nullable = true) var logo: String? = null
    var dataLibraryApiCode: Int? = null

    @OneToMany(mappedBy = "region")
    lateinit var cities: List<City>
}

enum class Country {
    KOREA
}